offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/offergeneration/QuotationDetails.groovy

```diff
@@ -93,19 +102,43 @@ class QuotationDetails {
     private void generateProductTable(List<OfferItem> items, ProductGroup productGroup, double subTotal) {
         //Create the items in html in the overview table
         //Check if there are ProductItems stored in list
+
+        String productGroupTableId = "${productGroup.toString()}-table"
+
         if (!items) {
-            htmlContent.getElementById("${productGroup.toString()}-table").remove()
+            htmlContent.getElementById(productGroupTableId).remove()
             return
         }
+
         //Clear Template content
         String elementId = generateElementID(0, productGroup)
-        htmlContent.getElementById(elementId).empty()
+        htmlContent.getElementById(productGroupTableId).empty()
 
-        if (isOverflowingPage()) {
-            generateHTMLTableOnNextPage("${productGroup.toString()}-table")
+        //Account for space necessary for Table Title
+        consumedPageSpace += TableElementSpacing.TABLE_TITLE.getTableElementSpacing()
+
+        //Account for Table Header space
+        consumedPageSpace += TableElementSpacing.TABLE_HEADER.getTableElementSpacing()
+
+        //Checks if the space allocated for the table header, title and the first item would overflow the current table and adds a pageBreak if necessary
+        if (isTableOverflowByItem(items.first())) {
+            generateHTMLTableOnNextPage(productGroupTableId)
             resetPageItemsCount()
+
+            //Account for space necessary for Table Title on new Page
+            consumedPageSpace += TableElementSpacing.TABLE_TITLE.getTableElementSpacing()
+
+            //Account for Table Header space on new Page
+            consumedPageSpace += TableElementSpacing.TABLE_HEADER.getTableElementSpacing()
```

Comment:
I would expect this to be not within the if statement, you need to add it anyhow. Rather I would opt for isTableOverflown to get the values of the space for header + title + first item and then decide for a page break or not. Then you can add the space for title and header. Regardless of a previous page break or not

Reply:
Good idea 👍